app/parse.py

```diff
@@ -15,8 +23,105 @@ class Product:
     num_of_reviews: int
 
 
+class Driver:
+    _instance = None
+
+    def __new__(cls, *args, **kwargs) -> "Driver":
+        if cls._instance is None:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self) -> None:
+        options = Options()
+        options.add_argument("--headless")
+        options.add_argument("--disable-gpu")
+        self.driver = webdriver.Chrome(options=options)
+
+    def get_driver(self) -> WebDriver:
+        return self.driver
+
+
+_driver = Driver().get_driver()
+
+pages = [
+    {"name": "home", "url": BASE_URL},
+    {"name": "computers", "url": urljoin(BASE_URL, "computers")},
+    {"name": "laptops", "url": urljoin(BASE_URL, "computers/laptops")},
+    {"name": "tablets", "url": urljoin(BASE_URL, "computers/tablets")},
+    {"name": "phones", "url": urljoin(BASE_URL, "phones")},
+    {"name": "touch", "url": urljoin(BASE_URL, "phones/touch")}
+]
+
+
+def parse_single_product(product_card: WebElement) -> Product:
+    title = product_card.find_element(By.CLASS_NAME,
+                                      "title").get_attribute("title")
+    description = product_card.find_element(By.CLASS_NAME, "description").text
+    price = float(product_card.find_element(By.CLASS_NAME, "price").text[1:])
```

Comment:
The price extraction assumes that the price string always starts with a currency symbol followed by the amount. Ensure that this assumption holds for all product cards, or consider adding error handling for unexpected formats.